cubi_tk/sodar/pull_raw_data.py

```diff
@@ -200,20 +220,24 @@ def _executed_commands(self, commands):
                 if answer_str.startswith("y") or answer_str.startswith("n"):
                     break
             answer = answer_str == "y"
+        failed_libs = []
         if not answer:
             logger.info("Answered 'no': NOT pulling files")
         else:
-            for cmd in commands:
+            for lib, cmd in commands.items():
                 try:
                     cmd_str = " ".join(map(shlex.quote, cmd))
                     logger.info("Executing %s", cmd_str)
                     print(cmd)
                     print(cmd_str)
                     check_call(cmd)
                 except SubprocessError as e:  # pragma: nocover
-                    logger.error("Problem executing irsync: %s", e)
-                    return 1
-        return 0
+                    if not self.config.allow_missing:
+                        logger.error("Problem executing irsync: %s", e)
+                        return [lib]
+                    else:
```

Comment:
I only thought about it after I give you the suggestion, but to keep it robust you need to include the exact error code from iRODS (there are many reasons the transfer would fail).
```

   if self.config.allow_missing and e.returncode = <iRODS error code you are expecting>:
        <warning messages logic>
   else:
          logger.error("Problem executing irsync: %s", e)
          return 1
   ...
```